src/logger.ts

```diff
@@ -8,25 +8,43 @@ import { PutLogEventsRequest } from 'aws-sdk/clients/cloudwatchlogs';
 import {
 	AWS_REGION,
 	CLIENT_LOGS_GROUP_NAME,
-	DIFF_SOURCE
+	DIFF_SOURCE,
+	LOG_AFTER_X_TIMES
 } from './constants';
 import { readYML, isEmpty } from './utils/common';
 import { generateSettings } from "./settings";
 
-let cloudwatchlogs = <AWS.CloudWatchLogs>{};
-
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-ignore
 const VERSION = vscode.extensions.getExtension('codesync.codesync').packageJSON.version;
 
-export function putLogEvent(error: string, userEmail?: string, additionalMsg="", retryCount=0) {
-	let errorMsg = error;
+let cloudwatchlogs = <AWS.CloudWatchLogs>{};
+const logErrorMsgTypes = {
+	CRITICAL: "CRITICAL",
+	ERROR: "ERROR",
+	WARNING: "WARNING",
+	INFO: "INFO",
+	DEBUG: "DEBUG"
+};
+
+export class CodeSyncLogger {
+	static info (msg: string, userEmail?: string, additionalMsg="", retryCount=0) {
+		putLogEvent(msg, logErrorMsgTypes.INFO, userEmail, additionalMsg, retryCount);
+	}
+
+	static error (msg: string, userEmail?: string, additionalMsg="", retryCount=0) {
```

Comment:
Can you add a `critical` message type as well for errors that inhibit the successful operation of plugin and are blocking for users.

Reply:
The error types that I used with my understanding are.

`DEBUG`: for developer oriented messages, only usable for brief testing of new features and should be removed once new features are fully tested.
`INFO`: Informational messages, could be useful while debugging but can be ignore during normal execution.
`WARNING`: Mild errors that do not affect the user but should be fixed in some time frame
`ERROR`: Errors that cause a bad UX and should be fixed soon.
`CRITICAL`: Errors that are blocking for the normal operation of the plugin and should be fixed immediately.

I would like your input on the above, once these are finalised I will add these in the documentation as well.

Also, I have used error types in all uppercase.

Reply:
👍 I agree. Let me add critical type as well. 
I am already using error types in upper case [here](https://github.com/codesyncapp/codesync-vs/pull/22/files#diff-7bc1140e724e284ae22084e1172568baa8d151063d02c75ed1d8805fa7796acaR22) if that's what you meant